qst/src/main/java/io/deephaven/qst/type/CustomType.java

```diff
@@ -43,10 +43,17 @@ final void checkClazz() {
     }
 
     @Check
-    final void checkType() {
+    final void checkNotArray() {
         if (clazz().isArray()) {
             throw new IllegalArgumentException(String.format(
                 "Can't create an array type here, use '%s' instead", NativeArrayType.class));
         }
     }
+
+    @Check
+    final void checkNotDbArray() {
+        if (clazz().getName().startsWith("io.deephaven.db.tables.dbarrays.Db")) {
```

Comment:
`!DbArrayBase.class.isAssignableFrom(clazz())` would be a better test.

Reply:
DbArrayBase is not on the classpath for QST.